relay-server/src/services/health_check.rs

```diff
@@ -182,127 +208,61 @@ impl HealthCheckService {
         }
     }
 
-    async fn handle_is_healthy(&self, message: IsHealthy) -> Status {
-        let upstream = self.upstream_relay.clone();
-
-        if self.config.relay_mode() == RelayMode::Managed {
-            let fut = upstream.send(IsNetworkOutage);
-            tokio::spawn(async move {
-                if let Ok(is_outage) = fut.await {
-                    metric!(gauge(RelayGauges::NetworkOutage) = u64::from(is_outage));
-                }
-            });
-        }
-
-        if matches!(message, IsHealthy::Liveness) {
-            return Status::Healthy;
-        }
-
-        if self.is_shutting_down.load(Ordering::Relaxed) {
-            return Status::Unhealthy;
-        }
+    async fn check_readiness(&mut self) -> Status {
+        // System memory is sync and requires mutable access, but we still want to log errors.
+        let sys_mem = self.system_memory_probe();
 
-        let (sys_mem, auth, agg, proj) = tokio::join!(
-            self.probe("system memory", self.system_memory_probe()),
+        let (sys_mem, auth, agg, proj, _) = tokio::join!(
+            self.probe("system memory", async { sys_mem }),
             self.probe("auth", self.auth_probe()),
             self.probe("aggregator", self.aggregator_probe()),
             self.probe("spool health", self.spool_health_probe()),
+            self.probe("network outage", self.network_outage_probe()),
         );
 
         Status::from_iter([sys_mem, auth, agg, proj])
     }
-
-    async fn handle_message(&self, message: HealthCheck) {
-        let HealthCheck(message, sender) = message;
-
-        let ty = message.variant();
-        let response = relay_statsd::metric!(
-            timer(RelayTimers::HealthCheckDuration),
-            type = ty,
-            { self.handle_is_healthy(message).await }
-        );
-
-        sender.send(response);
-    }
 }
 
 impl Service for HealthCheckService {
     type Interface = HealthCheck;
 
-    fn spawn_handler(self, mut rx: relay_system::Receiver<Self::Interface>) {
-        let service = Arc::new(self);
+    fn spawn_handler(mut self, mut rx: relay_system::Receiver<Self::Interface>) {
+        let (update_tx, update_rx) = watch::channel(StatusUpdate::healthy());
```

Comment:
Should we initialize with unhealthy?

Reply:
That would make sense to me. We'll have to make sure that our health checks are configured in a way that they allow a start up container to be unhealthy for up to 3 seconds.

Reply:
First health check runs immediately